TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/workflow/services/script_runner.py

```diff
@@ -0,0 +1,391 @@
+"""Script runner service for executing workflow scripts."""
+
+import json
+import logging
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Optional, Dict, Any
+
+logger = logging.getLogger(__name__)
+
+
+class ScriptRunner:
+    """Runs workflow scripts with standard interface."""
+
+    def __init__(self, script_dir: Optional[Path] = None):
+        """
+        Initialize with script directory path.
+
+        Args:
+            script_dir: Optional path to scripts. If None, uses CONFIG_PROCESSOR_DIR environment variable.
+                       Raises ValueError if neither is provided or if the directory doesn't exist.
+        """
+        if script_dir is None:
+            config_processor_dir = os.environ.get('CONFIG_PROCESSOR_DIR')
```

Comment:
nit: It might be better to not have a default script_dir - or maybe break this class into two - one to run scripts and another to run scripts from a specific place.  I don't want to need to put many different kinds of scripts all into the same directory.